modules/internal/String.chpl

```diff
@@ -63,12 +63,23 @@
 
   .. note::
 
-    While :record:`string` is intended to be a Unicode string, there is much
-    left to do. As of Chapel 1.17, only ASCII strings can be expected to work
-    correctly with all functions.
+    As of Chapel 1.18, :record:`string` is a multibyte string whose
+    character set is determined by the user's environment variable
+    settings.  This is normally UTF-8.  ASCII strings are a simple
+    subcase of UTF-8 strings, because every ASCII character is a UTF-8
+    character with the same meaning.
 
-    Future work involves support for both ASCII and unicode strings, and
-    allowing users to specify the encoding for individual strings.
+    For testing, we use the following settings.
+
+    ``export LANG=en_US.UTF-8``
+
+    ``export LC_COLLATE=C``
+
+    ``unset LC_ALL``
+
+    LANG sets the default character set.  LC_COLLATE overrides it for
+    sorting, so that we get consistent results.  Anything in LC_ALL
+    would override everything, so we unset it.
```

Comment:
This is OK at saying what the implementation does but I think it should start by saying what users should expect. But I'm not sure we know that yet. E.g. I would want the "normal" Chapel environment to be UTF-8.

Reply:
Yes, this documents the current situation.  It starts by saying that the user selects the character set with the environment variables, and then gives an example based on how we test things.  We could certainly decide to force UTF-8 in the future.

Reply:
Here is what I would recommend:

1. Look at the rendered string docs and fix remaining issues. Here are the issues I see:
   * there is a warning about casts from string to numeric types but no introduction that these are supported. Needs at least a 1-sentence description of what they do.
   * `record codePointIndex` is not documented

2. Let's introduce what the general idea is before talking about the current details. E.g. say "Chapel supports multibyte strings and is normally tested with UTF-8." and then go on to say the other stuff in a ..note.

3. We're missing docs about lengths and offsets in unicode strings.

I have created PR #11059 to make these changes. Please have a look.